x/logic/predicate/util.go

```diff
@@ -18,23 +23,76 @@
 	})
 }
 
+// IterMap transforms the output of an iterator by applying a given mapping function to each element.
+func IterMap[U any, V any](next func() (U, bool), f func(U) V) func() (V, bool) {
+	return func() (V, bool) {
+		u, ok := next()
+		if !ok {
+			var zeroV V
+			return zeroV, false
+		}
+
+		return f(u), true
+	}
+}
+
+// Accounts returns an iterator that iterates over all accounts.
+func Accounts(ctx context.Context, authQueryService types.AuthQueryService) func() (lo.Tuple2[sdk.AccountI, error], bool) {
+	var (
+		finished bool
+		key      []byte
+	)
+
+	return func() (lo.Tuple2[sdk.AccountI, error], bool) {
+		if finished {
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: nil}, false
+		}
+		res, err := authQueryService.Accounts(ctx,
+			&auth.QueryAccountsRequest{
+				Pagination: &query.PageRequest{
+					Key:   key,
+					Limit: 1,
+				},
+			})
+		if err != nil {
+			finished = true
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: err}, true
+		}
+
+		if len(res.Accounts) == 0 {
+			finished = true
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: nil}, false
+		}
+
+		key = res.Pagination.NextKey
+
+		interfaceRegistry := ctx.Value(types.InterfaceRegistryContextKey).(cdctypes.AnyUnpacker)
+		var account sdk.AccountI
+		if err := interfaceRegistry.UnpackAny(res.Accounts[0], &account); err != nil {
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: err}, true
```

Comment:
I quite agree with the principle, but less so with the suggestion. I believe the better approach is to pass the value explicitly through a function argument, rather than indirectly through the context. This makes the dependency clearer and eliminates the need for type assertions. I will make this change.